From ad1a4de6763f085e2cb1b2cb21a0ade7e77b9da2 Mon Sep 17 00:00:00 2001
From: Mathieu Bivert <mathieu.bivert@intel.com>
Date: Tue, 30 Apr 2013 09:48:23 +0200
Subject: crashtool: log fw loading failure

BZ: 104695

Category: aosp improvement
Domain: CWS.BT-Core
Origin: internal
Upstream-Candidate: no, proprietary

Change-Id: Ice42706067f500de48c9a2cf8e86c19656f70a3d
Orig-Change-Id: If44e9f07232da18c56bf0940b08467991937b893
Signed-off-by: Mathieu Bivert <mathieu.bivert@intel.com>
Reviewed-on: http://android.intel.com:8080/105486
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/123808
Reviewed-by: Poignet, Cedric <cedric.poignet@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 src/hardware.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/src/hardware.c b/src/hardware.c
index bf46128..a4496be 100644
--- a/src/hardware.c
+++ b/src/hardware.c
@@ -45,6 +45,8 @@
 #include "userial_vendor.h"
 #include "upio.h"
 
+#include <lct.h>
+
 /******************************************************************************
 **  Constants & Macros
 ******************************************************************************/
@@ -860,6 +862,7 @@ check_local_name:
                         if ((hw_cfg_cb.fw_fd = open(tmp_path, O_RDONLY)) == -1)
                         {
                             ALOGE("vendor lib preload failed to open [%s]", tmp_path);
+                            lct_log(CT_EV_STAT, "cws.bt", "fw_error", 0, tmp_path);
                         }
                     }
                     else
@@ -867,6 +870,7 @@ check_local_name:
                         ALOGE( \
                         "vendor lib preload failed to locate firmware patch file" \
                         );
+                        lct_log(CT_EV_STAT, "cws.bt", "fw_error", 0, tmp_path);
                     }
                 }
 
@@ -1060,6 +1064,7 @@ check_local_name:
     if (is_proceeding == FALSE)
     {
         ALOGE("vendor lib fwcfg aborted!!!");
+        lct_log(CT_EV_STAT, "cws.bt", "fw_cfg", 0);
         if (bt_vendor_cbacks)
         {
             if (p_buf != NULL)
-- 
1.7.4.1

